tabpfn_client/client.py

```diff
@@ -113,23 +122,37 @@ def predict(self, train_set_uid: str, x_test):
             ])
         )
 
-        self.error_raising(response, "predict")
+        self._validate_response(response, "predict")
 
         return np.array(response.json()["y_pred"])
 
-    def error_raising(self, response, method_name):
-        if response.status_code != 200:
-            load = None
-            try:
-                load = response.json()
-            except Exception:
-                pass
+    @staticmethod
+    def _validate_response(response, method_name, only_version_check=False):
+        # If status code is 200, no errors occurred on the server side.
+        if response.status_code == 200:
+            return
+
+        # Read response.
+        load = None
+        try:
+            load = response.json()
+        except Exception:
```

Comment:
Sorry I forgot to mention this, could we only catch known `Exception` and log the non-expected one? It seems a little dangerous.

I got the following from ChatGPT:
```python
try:
    response_payload = response.json()
except json.JSONDecodeError as e:
    logging.error(f"Failed to parse JSON from response in {method_name}: {e}")
```

Reply:
Remember to log it.